draft-gpew-priv-ppm.md

```diff
@@ -1004,21 +1123,137 @@ fetch certificates: https://datatracker.ietf.org/doc/html/rfc8555#section-7.4.2]
 issue#57). For example, how does a leader respond to a collect request if the
 helper drops out?]
 
-The collector then decrypts each aggregate share `share` using the following
-procedure:
+### Collection Aggregation {#collect-aggregate}
+
+The leader obtains each helper's encrypted aggregate share in order to respond
+to the collector's collect response. To do this, the leader first computes a
+checksum over the set of output shares included in the batch window identified
+by the collect request. The checksum is computed by taking the SHA256 hash of
+each nonce from the client reports included in the aggregation, then combining
+the hash values with a bitwise-XOR operation.
+
+Then, for each aggregator endpoint `[aggregator]` in the parameters associated
+with `CollectReq.task_id` (see {{collect-flow}}) except its own, the leader sends
+a POST request to `[aggregator]/aggregate_share` with the following message:
+
+~~~
+struct {
+  TaskID task_id;
+  Interval batch_interval;
+  uint64 report_count;
+  opaque checksum[32];
+  opaque helper_state<0..2^16>;
+} AggregateShareReq;
+~~~
+
+* `task_id` is the task ID associated with the PPM parameters.
+* `batch_interval` is the batch interval of the request.
+* `report_count` is the number of reports included in the aggregation.
+* `checksum` is the checksum computed over the set of client reports.
+* `helper_state` is the helper's state, which is carried across requests from
+  the leader.
+
+To handle the leader's request, the helper first ensures that the request meets
+the requirements for batch parameters following the procedure in
+{{batch-parameter-validation}}. If the batch parameters are invalid, then it
+MUST abort with error "batchInvalid".
+
+Next, it computes a checksum based on its view of the output shares included in
+the batch window, and checks that the `report_count` and `checksum` included in
+the request match its computed values. If not, then it MUST abort with error
+"batchMismatch".
```

Comment:
Should we add this to the errors table above? (I see it says "this list is not exhaustive", but I'm not sure if that means other errors may be in the same URI namespace, or other errors would have entirely different URIs)

Reply:
:+1:

Reply:
Yep, done.